tests/utils/managed_utils/mod.rs

```diff
@@ -0,0 +1,583 @@
+            self.deliver_welcome(welcome, group)?;
+        }
+        Ok(())
+    }
+
+    /// Has the `remover` remove the `target_members` from the Group `group`. If
+    /// the `remover` or one of the `target_members` is not part of the group,
+    /// it returns an error.
+    pub fn remove_clients(
+        &self,
+        action_type: ActionType,
+        group: &mut Group,
+        remover_id: &Vec<u8>,
+        target_members: Vec<Vec<u8>>,
+    ) -> Result<(), SetupError> {
+        let clients = self.clients.borrow();
+        let remover = clients
+            .get(remover_id)
+            .ok_or(SetupError::UnknownClientId)?
+            .borrow();
+        if group
+            .members
+            .iter()
+            .find(|(_, id)| {
+                target_members
+                    .iter()
+                    .find(|&client_id| client_id == id)
+                    .is_some()
+            })
+            .is_none()
+        {
+            return Err(SetupError::ClientNotInGroup);
+        }
+        let members = remover.get_members_of_group(&group.group_id)?;
+        let mut target_indices = Vec::new();
+        for target in &target_members {
+            let (index, _) = members
+                .iter()
+                .find(|(_, credential)| credential.identity() == target)
+                .unwrap();
+            target_indices.push(*index);
+        }
+        let mut remover_groups = remover.groups.borrow_mut();
+        let remover_group_state = remover_groups
+            .get_mut(&group.group_id)
+            .ok_or(SetupError::ClientNotInGroup)?;
+        let (messages, welcome_option) = match action_type {
+            ActionType::Commit => remover_group_state.remove_members(target_indices.as_slice())?,
+            ActionType::Proposal => {
+                let messages =
+                    remover_group_state.propose_remove_members(target_indices.as_slice())?;
+                (messages, None)
+            }
+        };
+        drop(remover_group_state);
+        drop(remover_groups);
+        drop(remover);
+        drop(clients);
+        self.distribute_to_members(&remover_id, group, messages)?;
+        if let Some(welcome) = welcome_option {
+            self.deliver_welcome(welcome, group)?;
+        }
+        Ok(())
+    }
+
+    pub fn perform_random_operation(&self, group: &mut Group) -> Result<(), SetupError> {
+        // Who's going to do it?
+        let member_id = group.random_group_member();
+        println!("Member performing the operation: {:?}", member_id);
+
+        // TODO: Do both things.
+        let action_type = match (OsRng.next_u32() as usize) % 2 {
+            0 => ActionType::Proposal,
+            1 => ActionType::Commit,
+            _ => return Err(SetupError::Unknown),
+        };
+
+        // TODO: Do multiple things.
+        let operation_type = match (OsRng.next_u32() as usize) % 3 {
+            0 => OperationType::Update,
```

Comment:
If you don't use the enum anywhere else, this looks a little overcomplicated. Why not just match the random number?

Reply:
I used to match the random number, but I thought using the enum would make it a bit more readable. If you prefer matching the number, I'm happy to restore that.

Reply:
Done!